bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/AbstractHomekitAccessoryImpl.java

```diff
@@ -267,29 +268,38 @@ protected <T> T getAccessoryConfiguration(HomekitCharacteristicType characterist
                 .map(homekitTaggedItem -> homekitTaggedItem.getConfiguration(key, defaultValue)).orElse(defaultValue);
     }
 
-    /**
-     * update mapping with values from item configuration.
-     * it checks for all keys from the mapping whether there is configuration at item with the same key and if yes,
-     * replace the value.
-     *
-     * @param characteristicType characteristicType to identify item
-     * @param map mapping to update
-     * @param customEnumList list to store custom state enumeration
-     */
     @NonNullByDefault
-    public <T> void updateMapping(HomekitCharacteristicType characteristicType, Map<T, String> map,
-            @Nullable List<T> customEnumList) {
-        getCharacteristic(characteristicType).ifPresent(c -> {
-            final Map<String, Object> configuration = c.getConfiguration();
-            if (configuration != null) {
-                HomekitCharacteristicFactory.updateMapping(configuration, map, customEnumList);
-            }
-        });
+    protected <T extends Enum<T> & CharacteristicEnum> Map<T, String> createMapping(
+            HomekitCharacteristicType characteristicType, Class<T> klazz) {
+        return createMapping(characteristicType, klazz, null, false);
+    }
+
+    @NonNullByDefault
+    protected <T extends Enum<T> & CharacteristicEnum> Map<T, String> createMapping(
+            HomekitCharacteristicType characteristicType, Class<T> klazz, boolean inverted) {
+        return createMapping(characteristicType, klazz, null, true);
```

Comment:
Did you mean to omit inverted here?

```suggestion
        return createMapping(characteristicType, klazz, null, inverted);
```

Reply:
good catch